Quickstart MP native image has dependency on jar file

## Environment Details
* Helidon Version: 2.0.0-SNAPSHOT
* Helidon MP
* JDK version: 11
----------

## Problem Description
During testing for PR #1576 I ran into the following problem. The native image builds successfully but if you run the resulting executable without the existence of `target/helidon-quickstart-mp.jar`  you get the following error:
```
Exception in thread "main" javax.enterprise.inject.spi.DeploymentException: Failed to obtain index view
	at io.helidon.microprofile.openapi.OpenApiCdiExtension.registerOpenApi(OpenApiCdiExtension.java:119)
...
Caused by: java.io.IOException: Attempted to read from previously-located index file jar:file:/home/jdipol/GitHub/barchetta/helidon/examples/quickstarts/helidon-quickstart-mp/target/helidon-quickstart-mp.jar!/META-INF/jandex.idx but the file cannot be found
	at io.helidon.microprofile.openapi.OpenApiCdiExtension.existingIndexFileReader(OpenApiCdiExtension.java:163)
	at io.helidon.microprofile.openapi.OpenApiCdiExtension.indexView(OpenApiCdiExtension.java:147)
	at io.helidon.microprofile.openapi.OpenApiCdiExtension.registerOpenApi(OpenApiCdiExtension.java:112)
...
Caused by: java.nio.file.NoSuchFileException: /home/jdipol/GitHub/barchetta/helidon/examples/quickstarts/helidon-quickstart-mp/target/helidon-quickstart-mp.jar
	at sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
...
	at io.helidon.microprofile.openapi.OpenApiCdiExtension.existingIndexFileReader(OpenApiCdiExtension.java:159)
```

If you build `target/helidon-quickstart-mp.jar` then the native image starts correctly.

--- a/microprofile/openapi/pom.xml
+++ b/microprofile/openapi/pom.xml
@@ -72,7 +72,7 @@
                         </goals>
                         <phase>process-test-classes</phase>
                         <configuration>
-                            <classesDir>${project.build.directory}/test-classes}</classesDir>
+                            <classesDir>${project.build.directory}/test-classes</classesDir>
                             <indexName>other.idx</indexName>
                             <fileSets>
                                 <fileSet>

--- a/microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/OpenApiCdiExtension.java
+++ b/microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/OpenApiCdiExtension.java
@@ -69,8 +69,8 @@ public class OpenApiCdiExtension implements Extension {
         HelidonFeatures.register("OpenAPI");
     }
 
-    private final List<URL> indexURLs; // used only in early processing
     private final String[] indexPaths;
+    private final int indexURLCount;
 
     private final Set<Class<?>> annotatedTypes = new HashSet<>();
 
@@ -88,7 +88,8 @@ public class OpenApiCdiExtension implements Extension {
 
     OpenApiCdiExtension(String... indexPaths) throws IOException {
         this.indexPaths = indexPaths;
-        indexURLs = findIndexFiles(indexPaths);
+        List<URL> indexURLs = findIndexFiles(indexPaths);
+        indexURLCount = indexURLs.size();
         if (indexURLs.isEmpty()) {
             LOGGER.log(Level.INFO, () -> String.format(
                     "OpenAPI support could not locate the Jandex index file %s "
@@ -130,7 +131,7 @@ public class OpenApiCdiExtension implements Extension {
      * @param event {@code ProcessAnnotatedType} event
      */
     private <X> void processAnnotatedType(@Observes ProcessAnnotatedType<X> event) {
-        if (indexURLs.isEmpty()) {
+        if (indexURLCount == 0) {
             Class<?> c = event.getAnnotatedType()
                     .getJavaClass();
             annotatedTypes.add(c);
@@ -146,7 +147,7 @@ public class OpenApiCdiExtension implements Extension {
      * reading class bytecode from the classpath
      */
     public IndexView indexView() throws IOException {
-        return !indexURLs.isEmpty() ? existingIndexFileReader() : indexFromHarvestedClasses();
+        return indexURLCount > 0 ? existingIndexFileReader() : indexFromHarvestedClasses();
     }
 
     /**

--- a/microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/OpenApiCdiExtension.java
+++ b/microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/OpenApiCdiExtension.java
@@ -69,7 +69,8 @@ public class OpenApiCdiExtension implements Extension {
         HelidonFeatures.register("OpenAPI");
     }
 
-    private final List<URL> indexURLs;
+    private final List<URL> indexURLs; // used only in early processing
+    private final String[] indexPaths;
 
     private final Set<Class<?>> annotatedTypes = new HashSet<>();
 
@@ -86,6 +87,7 @@ public class OpenApiCdiExtension implements Extension {
     }
 
     OpenApiCdiExtension(String... indexPaths) throws IOException {
+        this.indexPaths = indexPaths;
         indexURLs = findIndexFiles(indexPaths);
         if (indexURLs.isEmpty()) {
             LOGGER.log(Level.INFO, () -> String.format(
@@ -155,13 +157,16 @@ public class OpenApiCdiExtension implements Extension {
      */
     private IndexView existingIndexFileReader() throws IOException {
         List<IndexView> indices = new ArrayList<>();
-        for (URL indexURL : indexURLs) {
+        /*
+         * Do not reuse the previously-computed indexURLs; those values will be incorrect with native images.
+         */
+        for (URL indexURL : findIndexFiles(indexPaths)) {
             try (InputStream indexIS = indexURL.openStream()) {
                 LOGGER.log(Level.CONFIG, "Adding Jandex index at {0}", indexURL.toString());
                 indices.add(new IndexReader(indexIS).read());
             } catch (IOException ex) {
                 throw new IOException("Attempted to read from previously-located index file "
-                        + indexURL + " but the file cannot be found", ex);
+                        + indexURL + " but the index cannot be found", ex);
             }
         }
         return indices.size() == 1 ? indices.get(0) : CompositeIndex.create(indices);

